src/node/http.ts

```diff
@@ -927,7 +927,29 @@ export class HttpServer {
     }
 
     // Must be authenticated to use the proxy.
-    route.provider.ensureAuthenticated(request)
+    if (!route.provider.authenticated(request)) {
+      // Attempt to determine if it's the user browsing the root and if so fall
```

Comment:
```suggestion
      // Attempt to determine if it's the user browsing the route and if so fall
```

Reply:
I mean root, for example `8080.mydomain.com` and not `8080.mydomain.com/some/page`. Maybe we should always redirect though even if it's not the root.

Reply:
Well I guess we can't do that safely, it would end up causing the problem in #2128 again on account of the whole not being able to detect HTML nonsense.